packages/headless/src/controllers/commerce/core/pagination/headless-core-commerce-pagination.ts

```diff
@@ -46,17 +56,30 @@ export interface PaginationState {
   totalPages: number;
 }
 
-export type PaginationControllerState = Pagination['state'];
+export interface PaginationOptions {
+  pageSize: number;
+}
 
 export interface CorePaginationProps {
   fetchResultsActionCreator: FetchResultsActionCreator;
+  options?: PaginationOptions;
 }
 
+export type PaginationProps = Omit<
+  CorePaginationProps,
+  'fetchResultsActionCreator'
+>;
+
+const optionsSchema = new Schema({
+  pageSize: new NumberValue({min: 0, required: false}),
```

Comment:
should the minimum be 0 or 1? A page of 0 products is pretty useless 😅

Reply:
We now follow the API limits in: https://github.com/coveo/ui-kit/pull/3773/commits/e922ba472f6547601ccb460c62972b2efbf9451e